### ISSUE TITLE
GoogleSignIn SDK on Apple Silicon (M1)

### ISSUE THREAD

**Issue by malcommac** [firebase/firebase-ios-sdk#7208]
Hi, I tried to build my project on an M1 machine after updating the Firebase SDK to `7.2.0-M1`. However I still get this linking error on GoogleSignIn SDK. 
I'm not sure this is the right place to report it but the project itself does not have a repo and I supposed it's part of the same family.

This is the error:

`
ld: in .../Pods/GoogleSignIn/Frameworks/GoogleSignIn.framework/GoogleSignIn(GIDEMMErrorHandler_3a47e13d8ca81b41e9cdb7ef5468004a.o), building for iOS Simulator, but linking in object file built for iOS, file '..../Pods/GoogleSignIn/Frameworks/GoogleSignIn.framework/GoogleSignIn' for architecture arm64
`

**Comment by paulb777** [comment]
@malcolmdeck Sorry about the trouble. GoogleSignIn has not yet added support for the simulator on ARM-based Macs.

**Comment by malcommac** [comment]
thank you for your info @paulb777 . Did you have a timeline for this or some place where I can stay updated?

**Comment by paulb777** [comment]
@malcommac Sorry I don't have a timeline. This has also been raised at https://developer.apple.com/forums/thread/668320. I'll leave this issue open and update when we get an update from the GoogleSignIn team.

**Comment by malcommac** [comment]
Thank you I'll continue to follow this thread

**Comment by akmarinov** [comment]
+1

**Comment by malcommac** [comment]
> +1
> 
> The GoogleSignIn SDK hasn't seen an update in more than an year, so it's not looking good...

Did you know if there are alternatives? I would check if it’s still possible by implementing it manually.

**Comment by ericagredo** [comment]
> > +1
> > The GoogleSignIn SDK hasn't seen an update in more than an year, so it's not looking good...
> 
> Did you know if there are alternatives? I would check if it’s still possible by implementing it manually.

Yeah I would like to know if it's possible to implement manually. This issue is killer.

**Comment by ivopintodasilva** [comment]
+1

**Comment by ricardomsm** [comment]
+1

**Comment by kientux** [comment]
> > +1
> > The GoogleSignIn SDK hasn't seen an update in more than an year, so it's not looking good...
> 
> Did you know if there are alternatives? I would check if it’s still possible by implementing it manually.

Yes there is. Google sign in is basically using OAuth so you can manually implementing it using URLSession and/or a WKWebView.

**Comment by malcommac** [comment]
> > > +1
> > > The GoogleSignIn SDK hasn't seen an update in more than an year, so it's not looking good...
> > 
> > 
> > Did you know if there are alternatives? I would check if it’s still possible by implementing it manually.
> 
> Yes there is. Google sign in is basically using OAuth so you can manually implementing it using URLSession and/or a WKWebView.

I’m not sure it’s possible anymore; google blocked login from embedded browsers in 2017; the only way is to jump in safari out of the app.
https://auth0.com/blog/google-blocks-oauth-requests-from-embedded-browsers/

**Comment by nitish-kasturia** [comment]
Workaround that I am using is running Xcode with Rosetta 2.
It's an issue with the pod running natively on Apple Silicon so for simulator and the SwiftUI canvas, it can be run through Rosetta.

**Comment by timothycosta** [comment]
The simulator runs terribly under Rosetta, at least on my machine. Google, can we please get an update for this? M1s have been out for a while now.

**Comment by malcommac** [comment]
Yeah absolutely. About three months are passed and there are no news, no dates, everything is obscure.

**Comment by natanrolnik** [comment]
@paulb777 are there any updates on this, or maybe a timeline? All our 3rd party code was updated and the only missing piece to be able to build for the iOS Simulator on M1 Macs is the Google SignIn SDK.

**Comment by kientux** [comment]
While waiting for Google team to update...

If GoogleSignIn is not an critical part of your app, and you just want to run and develop other parts of the app on M1, then you can create a Xcode configuration for only M1 Mac. The configuration can be used for exclude GoogleSignIn when build the app:

<img width="519" alt="image" src="https://user-images.githubusercontent.com/4200743/107359606-7661dc00-6b07-11eb-8a85-29b864c6d5fa.png">

- For CocoaPods install
```
# Include all other configurations except `Debug-M1`
pod 'GoogleSignIn', :configurations => ['Debug', 'Staging', 'Release']
```

- For manual install: Untick Target membership when add GoogleSignIn.framework into project. Remove it from `Link Binary with libraries` build phase.

Then in Build settings, add a flag to check if current configuration is able to import GoogleSignIn:

<img width="687" alt="image" src="https://user-images.githubusercontent.com/4200743/107359764-aad59800-6b07-11eb-9264-f8178bc65fc2.png">

(`FAT_FRAMEWORK_NOT_AVAILABLE` means we can't import GoogleSignIn fat framework)

Finally in code:

```swift
#if !FAT_FRAMEWORK_NOT_AVAILABLE
import GoogleSignIn
#endif

class GoogleSignInManager: NSObject {
    #if !FAT_FRAMEWORK_NOT_AVAILABLE
    private let gid: GIDSignIn = GIDSignIn.sharedInstance()
    #endif

    /// Simply do nothing on M1 Mac
    func signIn(presenter: UIViewController) {
        #if !FAT_FRAMEWORK_NOT_AVAILABLE
        self.presenter = presenter
        
        gid.delegate = self
        gid.presentingViewController = presenter
        gid.signIn()
        #endif
    }
}
```

Hope it helps someone! I'm currently using this method for my team to still able to developing on M1 while keep building the app for CI/CD on Intel mac mini.

**Comment by voidless** [comment]
> I'll leave this issue open and update when we get an update from the GoogleSignIn team.
@paulb777 
Hi, I have the same issue with GoogleTagManager (and its dependencies GoogleAnalytics, GoogleUtilitiesLegacy, GoogleSymbolUtilities).
Should I create a new issue here or is this one enough for all Google pods with vendored frameworks/libs?

**Comment by paulb777** [comment]
Sorry about the lack of visibility on GoogleSignIn. There has been some progress and I'll investigate what can be shared.

@voidless GoogleTagManager is another product from another team, so another issue would be helpful.

**Comment by goudsmit** [comment]
I've solved this for now by wrapping all GIDSignIn stuff in `#if !targetEnvironment(simulator)` blocks, similar to the solution above by @kientux but without a symbol.

**Comment by malcommac** [comment]
Any news?

**Comment by vince1393** [comment]
This is our last pod before we can get our app running on M1. We've already replaced google places because of the same issue, but unlikely we'll be able to do the same here. Hopefully this is fixed soon

**Comment by malcommac** [comment]
> Sorry about the lack of visibility on GoogleSignIn. There has been some progress and I'll investigate what can be shared.
> 
> @voidless GoogleTagManager is another product from another team, so another issue would be helpful.

Hi @paulb777 , did you get some insights about this issue? It should be pretty straightforward to produce a fat-binary version of the framework (or alternatively allows us to produce an alternative oauth based version). It's the only pod with a certain relevance which does not work on M1.

**Comment by paulb777** [comment]
Hi @malcommac I'm in the loop on the plans. There is progress, but sorry I can't talk about the details here. I'll update when there's news to share.

**Comment by MrJorgeXavier** [comment]
@malcommac, after I made the following changes in my project I could run my app with GoogleSignIn and other problematic pods on the simulator without problems, see if it helps you too:

In the podfile, add the following lines:
 
```ruby
post_install do |installer|
  installer.pods_project.build_configurations.each do |config|
    config.build_settings['VALID_ARCHS'] = 'arm64, arm64e, x86_64'
  end
end
```

In the project's Build Settings, go to Architectures > Excluded Architectures and add Any iOS Simulator SDK arm64.
  
![Captura de Tela 2021-03-19 às 11 23 25](https://user-images.githubusercontent.com/33957786/111795592-1ac50400-88a6-11eb-9a9e-d540f08dd39d.png)

**Comment by malcommac** [comment]
> @malcommac, after I made the following changes in my project I could run my app with GoogleSignIn and other problematic pods on the simulator without problems, see if it helps you too:
> 
> In the podfile, add the following lines:
> 
> ```ruby
> post_install do |installer|
>   installer.pods_project.build_configurations.each do |config|
>     if config.name != 'Release'
>       config.build_settings['VALID_ARCHS'] = 'arm64, arm64e, x86_64'
>     end
>   end
> end
> ```
> 
> In the project's Build Settings, go to Architectures > Excluded Architectures and add Any iOS Simulator SDK arm64.
> 
> ![Captura de Tela 2021-03-19 às 11 23 25](https://user-images.githubusercontent.com/33957786/111795592-1ac50400-88a6-11eb-9a9e-d540f08dd39d.png)

Works like a charm, thank you!

**Comment by timothycosta** [comment]
@malcommac @MrJorgeXavier Just FYI, this does allow you to build your project for the simulator, but it disables arm support there. I.e. you're now running your apps under Rosetta, not natively. It may also break UIScrollViews in your app (unless apple has fixed that).

**Comment by ashah023** [comment]
> @malcommac @MrJorgeXavier Just FYI, this does allow you to build your project for the simulator, but it disables arm support there. I.e. you're now running your apps under Rosetta, not natively. It may also break UIScrollViews in your app (unless apple has fixed that).

Thanks for the heads up. Btw, I enabled XCode to run under Rosetta and haven't had any issues with UIScrollViews, so far 🤞

**Comment by MrJorgeXavier** [comment]
@timothycosta thanks for the info. I imagined it might had this kind of side effect. Gladly I haven't had any issues with UIScrollViews, and I've been using this setting for quite a while now. Maybe Apple really fixed it.

**Comment by malcommac** [comment]
I imagined it might had this side effect but it’s better than having the entire environment running under Rosetta.
(UIScrollViews seems working fine here too).

**Comment by malcommac** [comment]
> @malcommac, after I made the following changes in my project I could run my app with GoogleSignIn and other problematic pods on the simulator without problems, see if it helps you too:
> 
> In the podfile, add the following lines:
> 
> ```ruby
> post_install do |installer|
>   installer.pods_project.build_configurations.each do |config|
>     if config.name != 'Release'
>       config.build_settings["VALID_ARCHS[sdk=iphonesimulator*]"] = "arm64 arm64e x86_64"
>     end
>   end
> end
> ```
> 
> __Edit: I put the "[sdk=iphonesimulator_]" setting because I was having problems to build for some physical devices. Now it's all fine._
> 
> In the project's Build Settings, go to Architectures > Excluded Architectures and add Any iOS Simulator SDK arm64.
> 
> ![Captura de Tela 2021-03-19 às 11 23 25](https://user-images.githubusercontent.com/33957786/111795592-1ac50400-88a6-11eb-9a9e-d540f08dd39d.png)

Unfortunately it seems fails to build the project on a device because it can't found linked frameworks.
For example: When I import the pod "A" in the main target, build process fails with 'Cannot found module A'.
Any idea?

**Comment by MrJorgeXavier** [comment]
> > In the podfile, add the following lines:
> > ...
> 
> @MrJorgeXavier Unfortunately it seems fails to build the project on a device because it can't found linked frameworks.
> For example: When I import the pod "A" in the main target, build process fails with 'Cannot found module A'.
> Any idea?

@malcommac That happened with me to after I cleaned my project. I edited my previous comment with the solution, what I did was changing the following lines:

```
config.build_settings['VALID_ARCHS'] = 'arm64, arm64e, x86_64'
```

to

```
config.build_settings["VALID_ARCHS[sdk=iphonesimulator*]"] = "arm64 arm64e x86_64"
```

This way the config only affects the simulator.

**Comment by malcommac** [comment]
> > > In the podfile, add the following lines:
> > > ...
> > 
> > 
> > @MrJorgeXavier Unfortunately it seems fails to build the project on a device because it can't found linked frameworks.
> > For example: When I import the pod "A" in the main target, build process fails with 'Cannot found module A'.
> > Any idea?
> 
> @malcommac That happened with me too after I cleaned my project. I edited my previous comment with the solution, what I did was changing the following lines:
> 
> ```
> config.build_settings['VALID_ARCHS'] = 'arm64, arm64e, x86_64'
> ```
> 
> to
> 
> ```
> config.build_settings["VALID_ARCHS[sdk=iphonesimulator*]"] = "arm64 arm64e x86_64"
> ```
> 
> This way the config only affects the simulator.

Hi, thank you.
I've also tried this:

```sh
 if config.name != "Debug"
       config.build_settings['VALID_ARCHS'] = 'arm64, arm64e, armv7, armv6, i386, x86_64'
     end
```

but I'm not sure it's correct too.

**Comment by malcommac** [comment]
I've also added the remaining arch because it fails on my device.
Thank you very much for your support :)

```ruby
installer.pods_project.build_configurations.each do |config|
      if config.name != 'Release'
        config.build_settings["VALID_ARCHS[sdk=iphonesimulator*]"] = "arm64, arm64e, armv7, armv6, i386, x86_64"
      end 
    end
```

**Comment by Erumaru** [comment]
Any news?

**Comment by angelolloqui** [comment]
I have managed to make it work (I believe, although I am still testing it) by following the steps here:
https://bogo.wtf/arm64-to-sim.html
Note that this is a very hacky solution, updating the framework manually to support the new arch, but seems to work ok. In our case, we have kept the original for AppStore builds, but using the modified version for development. So far it is seems to be working ok.

**Comment by Erumaru** [comment]
Any news?

**Comment by AnkitaPundir** [comment]
Any update?

**Comment by dsouksavatd** [comment]
Hi guys,
I was face the same issues with google-signin and facebook-sdk on my M1 Macbook Pro

**Comment by dsouksavatd** [comment]
Hi guys,
I was face the same issues with google-signin and facebook-sdk on my M1 Macbook Pro
after spent couple days, finally I was found the solution for this (worked for me)

you have open your Xcode using Rosetta 

to run Xcode using Rosetta, go to Applications-> (right click Xcode Application) -> Get Info -> check Open with using Resetta, completely quite Xcode and open it again and try to rebuild your project, it work for me on Google Signin and Facebook SDK, hope it may help you

<img width="264" alt="Screenshot 2021-05-06 at 21 06 26" src="https://user-images.githubusercontent.com/2195503/117312121-04532600-aeaf-11eb-9ae2-c101a2c39861.png">

Thanks

**Comment by jc-paris** [comment]
> @malcommac Sorry I don't have a timeline. This has also been raised at https://developer.apple.com/forums/thread/668320. I'll leave this issue open and update when we get an update from the GoogleSignIn team.

@paulb777 any updates from the GoogleSignIn team ?

**Comment by ericagredo** [comment]
@jc-paris I had to get rid of Google Sign In so I could use Xcode M1 :( . Honestly, given the lack of maintenance from the Google Sign In team it might be smart to get rid of a low reliability repo  ¯\_(ツ)_/¯ .

**Comment by jc-paris** [comment]
> @jc-paris I had to get rid of Google Sign In so I could use Xcode M1 :( . Honestly, given the lack of maintenance from the Google Sign In team it might be smart to get rid of a low reliability repo ¯_(ツ)_/¯ .

That's our last resort: Not linking GoogleSignIn.framework when building for iOS simulator.

**Comment by VickyPrajapati24** [comment]
Any updates, I am stucked now. only googlesignin pod not working.

**Comment by Dimillian** [comment]
Hello, would be wonderful to have it working. Thanks.

**Comment by BrentMifsud** [comment]
is this library abandoned? Aside from firebase, it seems all google pods have not been updated in ages. And none of them even support SPM yet.

**Comment by superman9740** [comment]
After spending a ton of time on this, my advise is to just ditch the GoogleSignIn iOS SDK altogether.  There has been no movement on this issue, and to see people just drop sign in with google from their apps altogether because of this issue is pretty bad.  Fortunately, oAuth isn't that hard to deal with.  Frameworks like https://github.com/openid/AppAuth-iOS make it pretty simple.  No, you don't have to use WKWebView(Google blocked that option a while back).  Yes, you do lose the simplicity of a built in Google Sign In button, but, IMHO, to be able to ditch Cocoapods altogether in favor of an all-SPM setup was worth it.  Goodbye, GoogleSignIn SDK.  Hello, freedom!!  :)

**Comment by diegostamigni** [comment]
After @superman9740 mentioned it, I thought I'd give it a go.

Now, because my google sign-in integration relies behind Firebase Auth, I thought I would try to use the strategy as [the Twitter one](https://firebase.google.com/docs/auth/ios/twitter-login) and surprise surprise? It works perfectly fine. 

Just to re-iterate here, look at my Podfile after months ago I moved everything to SPM. Only things left? The broken Google library re M1s and related dependencies which had to follow it:
```
target '---' do
  platform :ios, '14.0'
  inhibit_all_warnings!

  pod 'Firebase/Auth'
  pod 'Firebase/Crashlytics'
  pod 'GoogleSignIn'
end
```

Now I successfully managed to de-integrate pods, and my Xcode project compiles perfectly fine on M1. Finally this is over. Thanks @superman9740 for making me realise that this was a possibility 😄 🙏

**Comment by acecilia** [comment]
@paulb777 I guess the release of the GoogleSignIn xcframework is delayed because they use Bazel for building, and [Bazel does not support arm64 for simulator yet](https://github.com/bazelbuild/bazel/issues/11628).

This is [the same issue they have in TensorFlowLite for iOS](https://github.com/tensorflow/tensorflow/issues/47400).

I managed to hack around a solution for putting together an xcframework of TensorFlowLite (with M1 support) that may be useful for GoogleSignIn. See [here](https://github.com/revolut-mobile/TensorFlowLiteC-binary) (Full disclosure: I did not manage to use the generated `TensorFlowLiteC.xcframework` in production yet, so I cant be 100% sure yet that it works)

**Comment by Dimillian** [comment]
As a note, Xcode 12.5 under Rosetta doesn't work on the new macOS 12 beta, so if you were relying on this in the meantimes, it might come to and end as apparently Apple don't plan to support it anymore. 
Link: https://developer.apple.com/forums/thread/681749

**Comment by We-Are-Empire** [comment]
An update would be great :)

**Comment by JohnnyTseng** [comment]
Are there any alternatives or workaround to this issue? 
Most of our project's third party dependency has been updated to support M1, with just GoogleSignIn support, we can ditch the exclude arm 64 simulator settings in Xcode

**Comment by paulb777** [comment]
Sorry for the delay. There is active internal progress and we'll update here as soon as M1 support becomes available for GoogleSignIn.

**Comment by jenish2311** [comment]
![image](https://user-images.githubusercontent.com/9266284/123106057-f0ab5100-d455-11eb-905b-bedd31dfe28a.png)

Apply this setting on Xcode, 

open Xcode, 
in Xcode open pod file Build setting, 

Add Exclude Architecture - arm64

Architecture must be in pod setting - Apple silicon, Intel

Close Xcode, 
Remove pod. lock file
pod install.

And it works!!!! 

I tried this it works...
Happy Coding!!!

**Comment by DVSoftware** [comment]
@jenish2311 that's not the solution, it's only a workaround, which was already mentioned.

**Comment by JohnnyTseng** [comment]
> ![image](https://user-images.githubusercontent.com/9266284/123106057-f0ab5100-d455-11eb-905b-bedd31dfe28a.png)
> 
> Apply this setting on Xcode,
> 
> open Xcode,
> in Xcode open pod file Build setting,
> 
> Add Exclude Architecture - arm64
> 
> Architecture must be in pod setting - Apple silicon, Intel
> 
> Close Xcode,
> Remove pod. lock file
> pod install.
> 
> And it works!!!!
> 
> I tried this it works...
> Happy Coding!!!

This is a workaround, plus, you don't need to open Xcode using Rosetta, 
you just need to set all of your targets including pods to exclude arm64 simulator slice.

**Comment by superman9740** [comment]
Then, of course, you won’t be able to debug using the simulator on Apple silicon. My advise:  ditch this framework entirely. There are far easier routes one can take that provide the same functionality.

**Comment by kientux** [comment]
☹️  Sorry to say this, but [AppAuth](https://github.com/openid/AppAuth-iOS) is working perfectly with Google OAuth. So instead of waiting for GoogleSignIn SDK to be updated, just switch to AppAuth and have full control over the OAuth flow.

**Comment by mattdamon108** [comment]
I'm afraid to say that AppAuth gives us a bit different experiences against GoogleSignIn SDK. It seems to use the webview to process the sign in with AppAuth. Instead, GoogleSignIn SDK is more like the native one.

**Comment by superman9740** [comment]
It took me less than two hours to make the switch to AppAuth, so it’s not that hard. Not being able to run on the sim is a deal breaker for a lot of folks. Are you saying you test after you release?  Lol. Also, googlesignin is the only framework I use that hasn’t added SPM support. Once I left it behind, no more cocoapods.

**Comment by JohnnyTseng** [comment]
> Then, of course, you won’t be able to debug using the simulator on Apple silicon. My advise: ditch this framework entirely. There are far easier routes one can take that provide the same functionality.

No, debug on Apple silicon using simulator is still possible, it's working pretty fine right now.

**Comment by BrentMifsud** [comment]
> It took me less than two hours to make the switch to AppAuth, so it’s not that hard. Not being able to run on the sim is a deal breaker for a lot of folks. Are you saying you test after you release?  Lol. Also, googlesignin is the only framework I use that hasn’t added SPM support. Once I left it behind, no more cocoapods. 

Also google places.

It seems googles closed source frameworks are pretty much ignored. They have not been updated in ages.

**Comment by mattdamon108** [comment]
Here is my workaround for my project running on simulator.

1. Add post install script in Podfile
```
# Podfile
post_install do |installer|
    # add this post install script -- start
    installer.pods_project.build_configurations.each do |config|
      config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] = "arm64"
    end
    # -- end
    react_native_post_install(installer)
end
```
2. pod install
```shell
$ pod install --repo-update
```

3. then run the script
```shell
$ yarn ios
```

**Comment by dicksonconsultingllc** [comment]
> Here is my workaround for my project running on simulator.
> 
> 1. Add post install script in Podfile
> 
> ```
> # Podfile
> post_install do |installer|
>     # add this post install script -- start
>     installer.pods_project.build_configurations.each do |config|
>       config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] = "arm64"
>     end
>     # -- end
>     react_native_post_install(installer)
> end
> ```
> 
> 1. pod install
> 
> ```shell
> $ pod install --repo-update
> ```
> 
> 1. then run the script
> 
> ```shell
> $ yarn ios
> ```

That won't work on Apple Silicon, as the simulator is considered ARM64, as has been mentioned in other previous comments.

**Comment by dicksonconsultingllc** [comment]
> > Then, of course, you won’t be able to debug using the simulator on Apple silicon. My advise: ditch this framework entirely. There are far easier routes one can take that provide the same functionality.
> 
> No, debug on Apple silicon using simulator is still possible, it's working pretty fine right now for my projects.

Then there must be some detail you're leaving out, as I just tried this and it fails.  Excluding Arm64 for simulator sdk ON APPLE SILICON won't link.  How could it?

**Comment by elliottwilliams** [comment]
Here's another hacky workaround: you can manually edit the arm64-apple-ios `GoogleSignIn` binary to make it appear to be built for arm64-ios-simulator: https://bogo.wtf/arm64-to-sim.html

My team is using this to unblock ourselves, but it's obviously an unsavory workaround, and liable to break in future Xcode or macOS versions.

**Comment by JohnnyTseng** [comment]
> > > Then, of course, you won’t be able to debug using the simulator on Apple silicon. My advise: ditch this framework entirely. There are far easier routes one can take that provide the same functionality.
> > 
> > 
> > No, debug on Apple silicon using simulator is still possible, it's working pretty fine right now for my projects.
> 
> Then there must be some detail you're leaving out, as I just tried this and it fails. Excluding Arm64 for simulator sdk ON APPLE SILICON won't link. How could it?

@diegostamigni Xcode will instead build x86_64 arch, and you'll have to also open the simulator in Rosetta
Maybe you can share your project settings?

**Comment by janmisar** [comment]
> ☹️ Sorry to say this, but [AppAuth](https://github.com/openid/AppAuth-iOS) is working perfectly with Google OAuth. So instead of waiting for GoogleSignIn SDK to be updated, just switch to AppAuth and have full control over the OAuth flow.

If you are too lazy to write it, we did it for you 🤗 We have created [OpenGoogleSignInSDK](https://github.com/AckeeCZ/OpenGoogleSignInSDK) 🎉
It's very lightweight library written Swift, with no dependencies and it mirrors original GoogleSignInSDK API, so it's very easy to replace original SDK in your existing project.
Feel free to use, give a feedback or contribute 💪

**Comment by Ashok28** [comment]
> > ☹️ Sorry to say this, but [AppAuth](https://github.com/openid/AppAuth-iOS) is working perfectly with Google OAuth. So instead of waiting for GoogleSignIn SDK to be updated, just switch to AppAuth and have full control over the OAuth flow.
> 
> If you are too lazy to write it, we did it for you 🤗 We have created [OpenGoogleSignInSDK](https://github.com/AckeeCZ/OpenGoogleSignInSDK) 🎉
> It's very lightweight library written Swift, with no dependencies and it mirrors original GoogleSignInSDK API, so it's very easy to replace original SDK in your existing project.
> Feel free to use, give a feedback or contribute 💪

Great work Honza, thank you, works great 👍

**Comment by DagAgren** [comment]
@mattdamon108 That is an unsupported hack, and it will apparently not work any longer under macOS 12.

**Comment by petea** [comment]
I'm happy to announce that next week we'll be making a substantive update to Google Sign-In for iOS that will provide Simulator support on M1 Macs in addition to other improvements.  It's taken longer than we would have liked, but the wait is nearly over.  I'll provide another update here as soon as the new release is available.

**Comment by BrentMifsud** [comment]
> I'm happy to announce that next week we'll be making a substantive update to Google Sign-In for iOS that will provide Simulator support on M1 Macs in addition to other improvements. It's taken longer than we would have liked, but the wait is nearly over. I'll provide another update here as soon as the new release is available.

SPM support?

**Comment by petea** [comment]
> SPM support?

Yes, Swift Package Manager will be supported as well.

**Comment by fassko** [comment]
@petea Hi there, do you know which day could be the release?

**Comment by petea** [comment]
Google Sign-In for iOS 6 has just been released and is now open source on GitHub!

https://github.com/google/GoogleSignIn-iOS

**Comment by paulb777** [comment]
And includes M1 support! Thanks @petea!

**Comment by acecilia** [comment]
Thank you! ❤️ 

@paulb777 are you planning to release a prebuilt binary of `Google Sign-In` as part of the next firebase release?

**Comment by paulb777** [comment]
@acecilia Yes, the prebuilt binary is planned to be part of the 8.4.0 binary distributions cc: @ryanwilson

**Comment by yemd** [comment]
@paulb777  @petea Thanks for your work! I am not sure how to use this new version tho? 
I tried pod update or directly specifying the version 6.0.0 (currently 5.0.2), updating cocoapods but nothing is working, any ideas?

**Comment by petea** [comment]
@yemd I've responded here: google/GoogleSignIn-iOS#26

**Comment by yemd** [comment]
> @yemd I've responded here: [google/GoogleSignIn-iOS#26](https://github.com/google/GoogleSignIn-iOS/issues/26)

Thank you, I sent you a Podfile.lock

**Comment by Meycon** [comment]
I have the same problem with Google Sign-in for React Native version where the pod requires version 5.0 and I can't change it to 6

**Comment by paulb777** [comment]
Yep - the conclusion at google/GoogleSignIn-iOS#26 is React Native support is dependent upon the resolution of react-native-google-signin/google-signin#948. Please follow along there.

**Comment by paulb777** [comment]
Closed for comments. Please do any additional followups at https://github.com/google/GoogleSignIn-iOS
